### DISCUSSION TITLE
Resolving VM names when using socket_vmnet network

### DISCUSSION THREAD

**Discussion by Ranjandas** [lima-vm/lima#2358] [Q&A] [answered]
Is there an option to resolve virtual machine names using DNS (similar to user-v2) when using the `socket_vmnet` network type?

**Comment by Ranjandas** [comment, marked as answer]
If anyone else has this requirement, I managed to get it working using mDNS. Initially, I tried with Avahi, but then later realized that `systemd-resolved` has a built-in option to use mDNS.

All I had to do is:

* Create the resolved config

```
# file: /etc/systemd/resolved.conf.d/mdns.conf
[Resolve]
MulticastDNS=yes
``` 
* Restart systemd-resolved

```
sudo systemctl restart systemd-resolved
```

Once this is done, other VMs using socket_vmnet can be addressed by using `<vm-name>.local`.

```
[ranjan@lima-murphy-srv-01 ~]$ systemd-resolve lima-murphy-cli-01.local
lima-murphy-cli-01.local: 192.168.105.10       -- link: lima0

-- Information acquired via protocol mDNS/IPv4 in 2.6ms.
-- Data is authenticated: no; Data was acquired via local or encrypted transport: no
-- Data from: cache
```

I hope this helps 🙏

**Reply by nirs** [reply]
Since we control the vm, this can be also be done via the provision scripts. Or maybe it is missing in the boot scripts since it works for the builtin user network.

**Comment by nirs** [comment]
@Ranjandas do you want to send a documentation change to make this more accessible for others?
